src/functions.c

```diff
@@ -616,6 +616,9 @@ uint64_t fcallGetCommandFlags(client *c, uint64_t cmd_flags) {
 }
 
 static void fcallCommandGeneric(client *c, int ro) {
+    /* Functions need to be fed to monitors before the commands they execute. */
+    replicationFeedMonitors(c,server.monitors,c->db->id,c->argv,c->argc);
```

Comment:
in theory, maybe the better place to put this is after the argument verification, just before `engine->call`, so that if the function is rejected either due to `Bad number of keys provided`, or the various checks in `scriptPrepareForRun` (e.g. `Can not run script with write flag on readonly replica`), it'll not get logged in MONITOR.

but since i see in EVAL we didn't do that, and since in these cases it'll probably be easy to see in the monitor that the script didn't actually call any redis command, i guess we can dismiss that concern.

arguing that the requirements for MONITOR are flexible, and it can log a command that was executed but failed.
e.g. commands that fail on OOM or arity will be missing, but ones who fail on WRONGTYPE will be logged.

Reply:
I did consider this but opted to keep it consistent with EVAL. I think monitor should eventually report if a command errors. Perhaps an area for a future improvement.

Reply:
yeah, we do have some issue listing many ideas for monitor, and various filters, one of them is to see only errors.
we'll deal with that then..